src/doca_cc_helper.h

```diff
@@ -289,14 +372,120 @@ cc_server_message_recv_callback(struct doca_cc_event_msg_recv *event, uint8_t *r
 	/* Save the connection that the ping was sent over for sending the response */
 	struct cc_ctx_server *ctx_server = (struct cc_ctx_server *)user_data.ptr;
 	ctx_server->ctx.connection = cc_connection;
-	
-	//DOCA_LOG_INFO("Message received: '%d, pointer is %p", (int)msg_len, recv_buffer);
+
+	DOCA_LOG_INFO("Message received: '%d, pointer is %p", (int)msg_len, recv_buffer);
 
 	memcpy(ctx_server->ctx.recv_buffer, recv_buffer, msg_len);
 	ctx_server->ctx.buf_size = (int)msg_len;
 	ctx_server->ctx.recv_flag = true;
 }
 
+/**
+ * Callback for consumer post recv task successfull completion
+ *
+ * @task [in]: Recv task object
+ * @task_user_data [in]: User data for task
+ * @ctx_user_data [in]: User data for context
+ */
+static void
+cc_consumer_recv_task_completion_callback(struct doca_cc_consumer_post_recv_task *task, union doca_data task_user_data,
+					union doca_data user_data)
+{
+	size_t recv_msg_len;
+	void *recv_msg;
+	struct doca_buf *buf;
+	doca_error_t result;
+
+	(void)task_user_data;
+	struct cc_ctx *ctx = (struct cc_ctx *)user_data.ptr;
+
+	buf = doca_cc_consumer_post_recv_task_get_buf(task);
+
+	result = doca_buf_get_data(buf, &recv_msg);
+	if (result != DOCA_SUCCESS) {
+		DOCA_LOG_ERR("Failed to get data address from DOCA buf with error = %s", doca_error_get_name(result));
+	}
+
+	result = doca_buf_get_data_len(buf, &recv_msg_len);
+	if (result != DOCA_SUCCESS) {
+		DOCA_LOG_ERR("Failed to get data length from DOCA buf with error = %s", doca_error_get_name(result));
+	}
+
+	ctx->buf_size = (int)recv_msg_len;
+	ctx->recv_flag = true;
+
+	DOCA_LOG_INFO("Message received: '%.*s'", (int)recv_msg_len, (char *)recv_msg);
+	(void)doca_buf_dec_refcount(buf, NULL);
+	doca_task_free(doca_cc_consumer_post_recv_task_as_task(task));
```

Comment:
no need for doca_buf_dec & free, can save doca_buf under fifo_ctx and use the same buf. 
need to verify when buf wasn't used during nonblock call